inmoose/sim/splat.py

```diff
@@ -0,0 +1,195 @@
+    dropout : bool, optional
+        if True, simulate scRNASeq data by adding drop-outs. Defaults to `False`.
+
+    alpha : float, optional
+        shape of the gamma distribution to draw the initial means from
+    beta : float, optional
+        rate of the gamme distribution to draw the initial means from
+    outlier_prob : float, optional
+        proportion of outliers
+    outlier_location : float, optional
+        location parameter for the log-normal distribution to draw the outlier
+        inflation factors from
+    outlier_scale : float, optional
+        scale parameter for the log-normal distribution to draw the outlier
+        inflation factors from
+    libsize_loc : float, optional
+        location of the log-normal distribution to draw the library sizes from
+    libsize_scale : float, optional
+        scale of the log-normal distribution to draw the library sizes from
+    phi : float, optional
+        common dispersion
+    bcv_df : int, optional
+        degrees of freedom in the inverse chi-square distribution used to draw
+        the biological coefficient of variation parameters
+    x0 : float, optional
+        dropout midpoint, used to draw the dropouts. Unused if
+        `single_cell=False`.
+    k : float, optional
+        dropout shape, used to draw the dropouts. Unused if `single_cell=False`.
+    random_state : {int, `numpy.random.Generator` or `numpy.random.RandomState`},
+                    optional
+        If `random_state` is None (or `np.random`), the
+        `numpy.random.RandomState` singleton is used.  If `random_state` is an
+        int, a new ``RandomState`` instance is used, seeded with
+        `random_state`.  If `random_state` is already a ``Generator`` or
+        ``RandomState`` instance, that instance is used.
+
+    Returns
+    -------
+    ndarray
+        simulated count matrix, of size `nb_genes` x `nb_samples`
+    """
+
+    # make sure we have a Generator
+    random_state = np.random.default_rng(random_state)
+
+    if batch is None:
+        batch = np.zeros(nb_samples, dtype=np.int64)
+    batch = np.asarray(batch)
+    assert len(batch) == nb_samples
+
+    if group is None:
+        group = np.zeros(nb_samples, dtype=np.int64)
+    group = np.asarray(group)
+    assert len(group) == nb_samples
+
+    nb_batches = len(np.unique(batch))
+    nb_groups = len(np.unique(group))
+
+    # draw gene means
+    original_means = gamma.rvs(alpha, beta, size=nb_genes, random_state=random_state)
+
+    # draw outliers
+    is_outlier = bernoulli.rvs(outlier_prob, size=original_means.shape, random_state=random_state)
+    outlier_factor = lognorm.rvs(outlier_location, outlier_scale, size=original_means.shape, random_state=random_state)
+    gene_means = np.where(is_outlier, outlier_factor * np.median(original_means), original_means)
+
+    # draw library sizes
+    L = lognorm.rvs(libsize_loc, libsize_scale, size=nb_samples, random_state=random_state)
```

Comment:
Same here, not sure about the parametrisation of scipy
scale seems to be the first argument (even if another argument is named scale), and loc should be at the loc argument
We can discuss that

Reply:
Got it.
In fact, R `rlnorm` is parameterized with the log-mean and log-sd, so the correct Python counterpart is `lognorm.rvs(s=scale, scale=exp(loc), size=...)`